backend/BackendMIIT/src/main/java/com/example/BackendMIIT/service/impl/DirectionServiceImpl.java

```diff
@@ -17,9 +20,34 @@
 public class DirectionServiceImpl implements DirectionService {
 
     private final DirectionRepository directionRepository;
+    private final DirectionMapper directionMapper;
 
-    public DirectionServiceImpl(DirectionRepository directionRepository) {
+    public DirectionServiceImpl(DirectionRepository directionRepository, DirectionMapper directionMapper) {
         this.directionRepository = directionRepository;
+        this.directionMapper = directionMapper;
+    }
+
+    @Override
+    public DirectionDto getDirectionByName(String name) {
+        Direction direction = directionRepository.findByName(name)
+                .orElseThrow(() -> new EntityNotFoundException("Direction doesn't exist"));
```

Comment:
If we didn't catch `EntityNotFoundException` the application is crushed?

Reply:
No, because parent of this exception is RuntimeException

Reply:
If RuntimeException didn't catch then the application is crushed